pages/_app.js

```diff
@@ -1,9 +1,28 @@
+import { useState } from "react";
 import GlobalStyle from "../styles";
-
+import FavoriteSpots from "../pages/favorites/index.js";
+import { mockSpots } from "../helpers/mockSpots.js";
 export default function App({ Component, pageProps }) {
+  const [spots, setSpots] = useState(mockSpots);
+
+  function toggleBookmark(id) {
+    const newSpotArray = spots?.map((spot) => {
+      if (spot.id === id) {
+        return {
+          ...spot,
+          isBookmark: !spot.isBookmark,
+        };
+      } else {
+        return spot;
+      }
+    });
+    setSpots(newSpotArray);
+  }
+
   return (
     <>
       <GlobalStyle />
+      <FavoriteSpots spots={spots} toggleBookmark={toggleBookmark} />
       <Component {...pageProps} />
```

Comment:
you can pass the sports and the toggleBookmark function down to the generic `<Component>` this makes it available on all pages.

I guess you want `<FavoriteSpots>` only on the favorites page, and not on all pages

```suggestion
      <Component {...pageProps} spots={spots} toggleBookmark={toggleBookmark} />
```